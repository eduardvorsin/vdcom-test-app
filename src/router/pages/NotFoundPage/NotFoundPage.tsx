import React from 'react';
import { Box, Typography, Link, styled } from '@mui/material';
import { Link as RouterLink } from 'react-router-dom';
import SdCardAlertOutlinedIcon from '@mui/icons-material/SdCardAlertOutlined';

const NotFoundWrapper = styled(Box)(() => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  height: '100vh',
  minHeight: '400px',
  padding: '0 15px',
}));

const NotFoundIcon = styled(SdCardAlertOutlinedIcon)(({ theme }) => ({
  width: '60px',
  height: '60px',
  marginBottom: '25px',

  [theme.breakpoints.up('sm')]: {
    width: '70px',
    height: '70px',
    marginBottom: '35px',
  },
  [theme.breakpoints.up('md')]: {
    width: '85px',
    height: '85px',
    marginBottom: '40px',
  },
  [theme.breakpoints.up('lg')]: {
    width: '100px',
    height: '100px',
    marginBottom: '50px',
  },
}));

const NotFoundLink = styled(Link)(() => ({
  display: 'inline-flex',
  alignItems: 'center',
  cursor: 'pointer',
  color: '#32302c',
  fontSize: '20px',
  textDecorationColor: 'currentColor',
  transition: 'color 0.15s ease',

  '&:hover': {
    color: '#EDBC4A',
    transition: 'color 0.15s ease',
  },
}));

const NotFoundTitle = styled(Typography)(({ theme }) => ({
  marginBottom: '20px',

  [theme.breakpoints.up('md')]: {
    marginBottom: '30px',
  },
  [theme.breakpoints.up('lg')]: {
    marginBottom: '40px',
  },
}));

const NotFoundText = styled(Typography)(({ theme }) => ({
  textAlign: 'center',
  fontSize: '22px',
  marginBottom: '10px',

  [theme.breakpoints.up('sm')]: {
    fontSize: '24px',
  },
  [theme.breakpoints.up('md')]: {
    fontSize: '26px',
    marginBottom: '15px',
  },
  [theme.breakpoints.up('lg')]: {
    fontSize: '30px',
    marginBottom: '20px',
  },
}));

const NotFoundPage = () => (
  <NotFoundWrapper
    data-testid='not-found-page'
  >
    <NotFoundIcon />

    <NotFoundTitle
      variant='h1'
      component='h1'
    >
      404
    </NotFoundTitle>
    <NotFoundText
      paragraph
    >
      The page you are looking for was not found
    </NotFoundText>
    <NotFoundLink
      to='/'
      component={RouterLink}
    >
      Go to the main page
    </NotFoundLink>
  </NotFoundWrapper>
);

export default NotFoundPage;
