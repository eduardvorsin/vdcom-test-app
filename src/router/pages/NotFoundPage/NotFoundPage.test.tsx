import React from 'react';

import { render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import NotFoundPage from './NotFoundPage';

const routes = [{
  path: '/',
  element: <NotFoundPage />,
}];

const router = createMemoryRouter(routes, {
  initialEntries: ['/'],
});

describe('NotFoundPage tests', () => {
  it('renders correctly', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByRole<HTMLTitleElement>('heading', { name: /404/ })).toBeInTheDocument();
    expect(screen.getByText<HTMLParagraphElement>(/The page you are looking for was not found/i)).toBeInTheDocument();
  });

  it('is a basic snapshot', () => {
    render(<RouterProvider router={router} />);

    expect(screen.getByTestId<HTMLDivElement>('not-found-page')).toMatchSnapshot();
  });
});
