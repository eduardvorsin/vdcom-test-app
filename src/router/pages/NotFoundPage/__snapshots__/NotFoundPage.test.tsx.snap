// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage tests is a basic snapshot 1`] = `
<div
  class="MuiBox-root css-1j9u8ue"
  data-testid="not-found-page"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1ngv1pw-MuiSvgIcon-root"
    data-testid="SdCardAlertOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M18 2h-8L4.02 8 4 20c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 18H6V8.83L10.83 4H18v16zm-7-5h2v2h-2zm0-7h2v5h-2z"
    />
  </svg>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1n9u69a-MuiTypography-root"
  >
    404
  </h1>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-erwkhi-MuiTypography-root"
  >
    The page you are looking for was not found
  </p>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-8qvnvx-MuiTypography-root-MuiLink-root"
    href="/"
  >
    Go to the main page
  </a>
</div>
`;
